composer/tools/copy_environment.py

```diff
@@ -0,0 +1,761 @@
+
+# Copyright 2018 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# [START composer_copy_environment]
+"""Script to create a copy of an existing Cloud Composer Environment.
+
+Creates a clone of a Composer Environment, copying Environment Configurations,
+DAGs/data/plugins/logs, and DAG run history. This script can be useful when
+migrating to new Cloud Composer releases.
+
+To use:
+* Upload to cloudshell
+* Run
+    `python copy_environment.py PROJECT LOCATION EXISTING_ENV_NAME NEW_ENV_NAME
+    [--running_as_service_account]`
+"""
+from __future__ import print_function
+
+import argparse
+import ast
+import base64
+import contextlib
+import json
+import os
+import platform
+import re
+import shutil
+import subprocess
+import sys
+import tempfile
+import time
+import uuid
+
+from cryptography import fernet
+import google.auth
+from google.cloud import storage
+from google.oauth2 import service_account
+from googleapiclient import discovery, errors
+from kubernetes import client, config
+from mysql import connector
+import six
+from six.moves import configparser
+
+DEFAULT_SCOPES = ["https://www.googleapis.com/auth/cloud-platform"]
+
+
+def parse_args():
+    parser = argparse.ArgumentParser(
+        description="Clone a composer environment."
+    )
+    parser.add_argument(
+        "project",
```

Comment:
May be harder because this is using a positional argument but consider using the environment variable, `GOOGLE_CLOUD_PROJECT` or `GCLOUD_PROJECT` as the default. It's [a convention we use in our other samples](https://github.com/GoogleCloudPlatform/python-docs-samples/blob/master/iot/api-client/manager/manager.py#L777).